### ISSUE TITLE
Retain/release leaks

### ISSUE THREAD

**Issue by alesanro** [dactober/ioscourses#1]
I found that in `AppDelegate.m` file creation of a window and viewController creates leak of those objects
```objective-c
self.window=[[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
viewController = [[MyTableViewController alloc] init];//
```
As far as properties you try to assign hold strong references to its objects then you need to release created instances after they were created (or autorelease).

**Comment by alesanro** [comment]
@dactober take a look

**Comment by dactober** [comment]
Здравствуйте, я же переопределил dealloc или нужно именно autorelease?

2017-03-21 15:42 GMT+03:00 Alex Rudyak <notifications@github.com>:

**Comment by alesanro** [comment]
Да, здесь нужет `autorelease` (при условии, что ты сразу присваиваешь значение в property), так как способ создания объекта изначально увеличивает счётчик ссылок на *+1*, а присваивание в property делает ещё раз *+1*

**Comment by dactober** [comment]
Понял, исправлю и проверю что с запуском, перед отправкой проект запускался

**Comment by alesanro** [comment]
Есть большая вероятность, что эти ресурсы есть локально, но не были закоммичены.

**Comment by dactober** [comment]
Тогда это проблема, т.к. моя виртуальная машина слетела =( и все что есть
от проекта висит на github

On Mar 21, 2017 3:58 PM, "Alex Rudyak" <notifications@github.com> wrote:

Есть большая вероятность, что эти ресурсы есть локально, но не были
закоммичены.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/dactober/ioscourses/issues/1#issuecomment-288070232>,
or mute the thread
